jest/defs.bzl

```diff
@@ -128,70 +155,117 @@ def jest_test(
-            snapshot_label = to_label(snapshot)
-            if snapshot_label.package != native.package_name():
-                msg = "Expected jest_test '{target}' snapshots to be in test target package '{jest_test_package}' but got '{snapshot}' in package '{snapshot_package}'".format(
-                    jest_test_package = native.package_name(),
-                    snapshot = snapshot_label,
-                    snapshot_package = snapshot_label.package,
-                    target = to_label(name),
-                )
-                fail(msg)
-            if not snapshot_label.name.endswith(snapshot_ext):
-                msg = "Expected jest_test '{target}' snapshots to be labels to source files ending with extension '{snapshot_ext}' but got '{snapshot}'".format(
-                    snapshot = snapshot_label,
-                    snapshot_ext = snapshot_ext,
-                    target = to_label(name),
-                )
-                fail(msg)
-
-            generated_snapshot = "{}{}".format(snapshot, GENERATED_SNAPSHOT_SUFFIX)
-            generated_snapshots = generated_snapshot
-            update_snapshots_files[snapshot] = generated_snapshot
-
-        # This is build target that builds reference snapshots for the write_source_files snapshot
-        # updater target below. Generated reference snapshots have a GENERATED_SNAPSHOT_SUFFIX
-        # suffix so they the write_source_files target below is able to specify both the source file
-        # snapshots (in `snapshots`) and the generated reference snapshots by label.
+        _jest_update_snapshots(
+            name = name,
+            config = config,
+            data = data,
+            tags = tags,
+            update_directory = update_directory,
+            snapshot_files = snapshot_files,
+            gen_snapshots_bin = gen_snapshots_bin,
+            snapshots_ext = snapshots_ext,
+            silent_on_success = silent_on_success,
+        )
+
+def _jest_update_snapshots(
+        name,
+        config,
+        data,
+        tags,
+        update_directory,
+        snapshot_files,
+        gen_snapshots_bin,
+        snapshots_ext,
+        silent_on_success):
+    for snapshot in snapshot_files:
+        snapshot_label = to_label(snapshot)
+        if snapshot_label.package != native.package_name():
+            msg = "Expected jest_test '{target}' snapshots to be in test target package '{jest_test_package}' but got '{snapshot_label}' in package '{snapshot_package}'".format(
+                jest_test_package = native.package_name(),
+                snapshot_label = snapshot_label,
+                snapshot_package = snapshot_label.package,
+                target = to_label(name),
+            )
+            fail(msg)
+        if not snapshot_label.name.endswith(snapshots_ext):
+            msg = "Expected jest_test '{target}' snapshots to be labels to source files ending with extension '{snapshots_ext}' but got '{snapshot}'".format(
```

Comment:
someone's slides in bazelcon had an "f string"
https://www.geeksforgeeks.org/formatted-string-literals-f-strings-python/
so we could stop repeating the vars as named parameters of the format method. I dunno what version of bazel starlark supports it...

Reply:
that will be nice to have